packages/grid/x-data-grid/src/hooks/features/rows/useGridRows.ts

```diff
@@ -109,6 +109,19 @@ export const useGridRows = (
     [apiRef],
   );
 
+  const getRowId = React.useCallback<GridRowApi['getRowId']>(
```

Comment:
I was wondering about the performance of this.

And if like AG Grid, it can make sense to have an intermediary data structure for caching https://www.ag-grid.com/javascript-data-grid/client-side-row-stages/#state-2-all-rows

Reply:
Which kind of performance hit are you worried about?

And I'm not sure how that structure applies here.

Reply:
The `apiRef.current` indirection might prevent optimization, but I think if we destructure the API method into a const before usage like `const getRowId = apiRef.current.getRowId` it becomes very very easy for the JITs to optimize it away with inlining, and the `if (getRowIdProp)` branch is also optimized away in most cases due to `getRowIdProp` being a const `undefined`.

But we can keep that for cases where it matters. In the case for this PR it doesn't matter enough.

Reply:
Oh actually it's for filtering, so it might matter. I'll benchmark at some point and decide from there.

Reply:
It would be nice to avoid, but let's start with correctness and we can benchmark performance again soon, I'm completing one more perf change and I'll be done with the work from #9120. Anyway I think we've gained enough performance lately to afford one more fcall in there :)